### ISSUE TITLE
[java]A false positive about ImmutableField and @UtilityClass

### ISSUE THREAD

**Issue by clearlove2077** [pmd/pmd#4547]
**Affects PMD Version:7.0.0-rc1** 

**Rule: ImmutableField**

Please provide the rule name and a link to the rule documentation:
<https://docs.pmd-code.org/latest/pmd_rules_java_design.html#immutablefield>

**Description:**
The first code sample uses the annotation `@UtilityClass` which can make the class `final` and make fields `static`. So the analysis results of the two cases should be equivalent because these two cases are equivalent. However, PMD reports a warning in the first sample and no warnings in the second one because PMD ignores  the annotation `@UtilityClass`. PMD should not report a warning in `Case1` like the `Case2`.

**Code Sample demonstrating the issue:**
**Case1**
```java
import lombok.experimental.UtilityClass;
@UtilityClass
class MyClass1 {
    private String[] foobar1 = new String[0];   //  report a warning
    public final String[] call1() {
        return foobar1;
    }
}
```
**Case2**
```java
final class MyClass2 {
    private static String[] foobar1 = new String[0];    //  report no warning
    public static final String[] call1() {
        return foobar1;
    }
}
```

**Expected outcome:**

PMD reports a violation at line 4 in the first case, but that's wrong. That's a false positive.

**Comment by clearlove2077** [comment]
Besides, I find another two rules **SingularField** and **UnusedAssignment** suffer from similar problems. `@UtilityClass` can add a `static` modifier to the field and PMD should not report warnings.

**Comment by oowekyala** [comment]
I think this is not an FP, but a FN. In both your code samples in the OP, the field may be made final, so ImmutableField should report it. `@UtilityClass` does not make fields final, only static.
